api/config/config.go

```diff
@@ -202,17 +203,20 @@ type ClusterConfig struct {
 type ImageBuilderConfig struct {
 	ClusterName                 string `validate:"required"`
 	GcpProject                  string
+	GcsAuthenticationType       string
+	GcsCredentialsJson          string
```

Comment:
How is this value expected to be supplied during deployment time? If using Workload Identity, do we really need this?

Btw I just remembered we have this implementation, also used by Merlin as `MlflowDeleteService`, for model version delete:
* https://github.com/caraml-dev/mlp/blob/main/api/pkg/artifact/artifact.go#L7
* https://github.com/caraml-dev/mlp/blob/main/api/pkg/client/mlflow/mlflow.go

Can we make use of this instead of `gsutil.go` ? We will need to add the `ReadArtifact` and `WriteArtifact` APIs to the service [here](https://github.com/caraml-dev/mlp/blob/main/api/pkg/artifact/artifact.go#L11).

Reply:
I'm not aware of mlp.artifact package. Prior to writing this new gsutil package, I only see this gsutil being used by standard transformer -- which we should refactor too later. Turned out, this mlp.artifact is indirect dependencies of mlflow service.
I've updated the mlp.artifact package (thanks for the quick review there) and update merlin implementation to use it.